app/models/manageiq/providers/kubernetes/container_manager/metrics_capture/hawkular_client_mixin.rb

```diff
@@ -2,18 +2,27 @@ module ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::Hawkul
   def hawkular_client
     require 'hawkular/hawkular_client'
     @client ||= Hawkular::Metrics::Client.new(
-      hawkular_entrypoint, hawkular_credentials, hawkular_options)
+        hawkular_entrypoint, hawkular_credentials, hawkular_options)
   end
 
   def hawkular_entrypoint
-    hawkular_endpoint = @ext_management_system.connection_configurations.hawkular.try(:endpoint)
-    hawkular_endpoint_empty = hawkular_endpoint.try(:hostname).blank?
     worker_class = ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCollectorWorker
-
-    URI::HTTPS.build(
-      :host => hawkular_endpoint_empty ? @ext_management_system.hostname : hawkular_endpoint.hostname,
-      :port => hawkular_endpoint_empty ? worker_class.worker_settings[:metrics_port] : hawkular_endpoint.port,
-      :path => worker_class.worker_settings[:metrics_path] || '/hawkular/metrics')
+    path = worker_class.worker_settings[:metrics_path] || '/hawkular/metrics'
+    hawkular_endpoint = @ext_management_system.connection_configurations.hawkular.try(:endpoint)
+    if hawkular_endpoint.try(:hostname).present?
+      URI::HTTPS.build(
+          :host => hawkular_endpoint.hostname,
+          :port => hawkular_endpoint.port,
+          :path => path
+      )
+    else
+      hawkular_route_host = ContainerRoute.find_by(:name => "hawkular-metrics").try(:spec).try(:host)
```

Comment:
You have to also search by the ems other wise you will might get a route from a different provider.